snapcraft/tests/test_pluginhandler.py

```diff
@@ -301,6 +301,19 @@ def test_mark_done_clears_later_steps(self):
                         os.path.exists(handler._step_state_file(later_step)),
                         'Expected later step states to be cleared')
 
+    def test_state_file_migration(self):
+        for step in common.COMMAND_ORDER:
+            shutil.rmtree(common.get_partsdir())
+            with self.subTest('{} step'.format(step)):
+                part_name = 'foo'
```

Comment:
just a very minor thing, this doesn't need to be nested in the context nor the loop